backend/src/org/commcare/suite/model/graph/Graph.java

```diff
@@ -141,26 +141,35 @@ private void evaluateConfiguration(Configurable template, ConfigurableData data,
     }
 
     /*
-     * Helper for evaluate. Looks at a single series.
+     * Helper for evaluate. Looks at all series.
      */
     private void evaluateSeries(GraphData graphData, EvaluationContext context) {
         try {
             for (XYSeries s : mSeries) {
+                Hashtable<String, Text> expandableConfiguration = new Hashtable<>();
+                for (Enumeration<String> e = s.getExpandableConfigurationKeys(); e.hasMoreElements();) {
+                    String key = e.nextElement();
+                    Text value = s.getConfiguration(key);
+                    if (value != null) {
+                        expandableConfiguration.put(key, value);
+                    }
+                }
+
                 Vector<TreeReference> refList = context.expandReference(s.getNodeSet());
                 SeriesData seriesData = new SeriesData();
                 EvaluationContext seriesContext = new EvaluationContext(context, context.getContextRef());
 
-                evaluateConfiguration(s, seriesData, seriesContext);
-                // Guess at names for series, if it wasn't provided
-                if (seriesData.getConfiguration("name") == null) {
-                    seriesData.setConfiguration("name", s.getY());
-                }
-                if (seriesData.getConfiguration("x-name") == null) {
-                    seriesData.setConfiguration("x-name", s.getX());
-                }
-
+                Hashtable<String, Vector<String>> expandedConfiguration = new Hashtable();
                 for (TreeReference ref : refList) {
                     EvaluationContext refContext = new EvaluationContext(seriesContext, ref);
+                    for (Enumeration<String> e = expandableConfiguration.keys(); e.hasMoreElements();) {
+                        String key = e.nextElement();
+                        if (!expandedConfiguration.containsKey(key)) {
+                            expandedConfiguration.put(key, new Vector<String>());
+                        }
+                        String value = expandableConfiguration.get(key).evaluate(refContext);
+                        expandedConfiguration.get(key).addElement(value);
```

Comment:
Seeing `expandableConfiguration` and expandedConfiguration` right next to each other it is hard to differentiate them. What about something like `rawConfig` and `expandedConfig`?